microsetta_admin/server.py

```diff
@@ -401,63 +402,80 @@ def per_sample_summary():
-
-        # if we are here then the user is querying using barcodes and we
-        # simply need to set up the query below to perform.
-        sample_barcodes = [sample_barcode, ]
     else:
-        # assume POST, since there are only two methods defined in route.
-        # if we are here, it is because the user is querying using an uploaded
-        # file containing sample names.
-        sample_barcodes, err = upload_util.parse_request_csv_col(request,
-                                                                 'file',
-                                                                 'sample_name')
-        if err is not None:
-            # there was an error. abort early.
-            return render_template('per_sample_summary.html',
-                                   resource=None,
-                                   projects=projects,
-                                   **build_login_variables(),
-                                   search_error=[{'error': err}])
+        search_field = request.form.get('search_field')
+        search_value = request.form.get('text_input')
+        uploaded_file = request.files.get('file')
 
-    # perform the main query.
-    payload = {'sample_barcodes': sample_barcodes}
-    status, result = APIRequest.post('/api/admin/account_barcode_summary?stri'
-                                     'p_sampleid=%s' % str(strip_sampleid),
-                                     json=payload)
+        search_values = []
 
-    if status == 200:
-        if result['partial_result'] is True:
-            unprocessed_barcodes = result['unprocessed_barcodes']
+        if uploaded_file:
+            file_content = io.TextIOWrapper(uploaded_file,
+                                            encoding='utf-8-sig')
+            csv_reader = csv.reader(file_content)
+
+            for row in csv_reader:
+                search_values.extend(row)
         else:
-            unprocessed_barcodes = None
-        resource = pd.DataFrame(result['samples'])
-        order = ['sampleid', 'project', 'account-email',
-                 'source-type', 'site-sampled', 'sample-date',
-                 'sample-time', 'sample-status', 'sample-received',
-                 'ffq-taken', 'ffq-complete', 'vioscreen_username']
-        order.extend(sorted(set(resource.columns) - set(order)))
-        resource = resource[order]
+            search_values = [search_value] if search_value else []
+
+        payload = {}
+        if search_field == 'sample_barcode':
+            payload['sample_barcodes'] = search_values
+        elif search_field == 'kit_id':
+            payload['kit_ids'] = search_values
+        elif search_field == 'email':
+            payload['emails'] = search_values
+        elif search_field == 'outbound_tracking':
+            payload['outbound_tracking_numbers'] = search_values
+        elif search_field == 'inbound_tracking':
+            payload['inbound_tracking_numbers'] = search_values
```

Comment:
Wouldn't it be more efficient to have the value on the form be what's being sent to the API and then just have `payload[search_field] = search_values`?

Reply:
Ah yes, definitely. Changing.